activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -278,13 +368,14 @@ def release_connection(owner_thread = ActiveSupport::IsolatedExecutionState.cont
       # connection will be properly returned to the pool by the code that checked
       # it out.
```

Comment:
I am not sure I follow the code 100%, but I think this part of the method comment is no longer correct:

> If no connection
> is already checked out by the current thread, a connection will be checked
> out from the pool, yielded to the block, and then returned to the pool when
> the block is finished.

A freshly checked-out connection seems to no longer be returned to the pool at the end of the block _unless_ the user has explicitly made it un-sticky while inside the `with_connection` block.

Reply:
No, it's correct. `lease.sticky` defaults to `false`, and the connection is returned if nothing flipped it to `true` inside the block.

Reply:
Oh right, sorry. I got confused about the similar method names `#connection_lease` and `#lease_connection` and thought the lease would _always_ be set to sticky in line 292.